enhancements/workload-partitioning/cpu-limits-workload-partitioning.md

```diff
@@ -0,0 +1,417 @@
+	return mgr.Set(cg.Resources)
+}
+```
+
+This will all be exposed to the user via the `toml` configuration.
+
+```toml
+[crio.runtime.workloads.management]
+activation_annotation = "target.workload.openshift.io/management"
+annotation_prefix = "resources.workload.openshift.io"
+resources = { "cpushares" = 0,  "cpuquota" = 0, "cpuset" = "0-1,52-53" }
+```
+
+### Admission Webhook
+
+The admission webhook will be updated to use the CPU limit resource information
+to add the annotations to the Pod for each container. Pods with cpu limits will
+no longer be ignored and their limits will correctly be represented with the
+`management.workload.openshift.io/cores` in the same way that `requests` are.
+Containers that do not include `limits` will not have a `cpuquota` set. We will
+be utilizing the helper function in
+[cm.MilliCPUToQuota](https://github.com/openshift/kubernetes/blob/0e0d15b865ffc36177dc8770b4723dc14476a630/pkg/kubelet/cm/helpers_linux.go#L58-L82)
+to correctly pass along the CPU quota to the container runtime, but we will
+maintain the `milli` value for the `requests.limits.cpu` value so as to not
+cause confusion by changing the unit integer value. That is an implementation
+detail that does not need to be exposed to the user.
+
+In the example below, The resulting Pod from the given Deployment will now
+correctly translate the `resources.limits.cpu` to
+`management.workload.openshift.io/cores` to `cpuquota`. The Pod will then be
+annotated to use the new `cpuquota` attribute for the `busybox` container for
+the container runtime to alter the Cgroup
+`resources.workload.openshift.io/busybox: '{"cpushares": 20, "cpuquota":
+3000}'`. Note, the `cpuquota` correctly reflects the `30m` in microsecends. When
+no CPU limits are specified then the old behavior will still be in effect where
+no `cpuquota` is included, `resources.workload.openshift.io/busybox-no-limits:
+'{"cpushares": 20}'`
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: busybox-deployment
+spec:
+    ...
+  template:
+    metadata:
+        ...
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+    spec:
+      containers:
+      - ...
+        name: busybox
+        resources:
+          requests:
+            cpu: 20m
+            memory: 50Mi
+          limits:
+            cpu: 30m
+            memory: 50Mi
+      - ...
+        name: busybox-no-limits
+        resources:
+          requests:
+            cpu: 20m
+            memory: 50Mi
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    resources.workload.openshift.io/busybox: '{"cpushares": 20, "cpuquota": 3000}'
+    resources.workload.openshift.io/busybox-no-limits: '{"cpushares": 20}'
+    target.workload.openshift.io/management: '{"effect":"PreferredDuringScheduling"}'
+...
+spec:
+  containers:
+  - ...
+    name: busybox
+    resources:
+      limits:
+        management.workload.openshift.io/cores: "30"
```

Comment:
I don't think this will work. For extended resources the request/limit must be the same, https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#consuming-extended-resources

I think it would be best to keep the existing behavior where the extended resource represents the request and simply add the extra field to the annotation for the limit.

Reply:
Ah,  I see, missed that. Updated to correctly reflect this